app/screens/apps_form/apps_form_container.tsx

```diff
@@ -0,0 +1,177 @@
+    constructor(props: Props) {
+        super(props);
+        this.state = {form: props.form};
+    }
+
+    handleSubmit = async (submission: {values: AppFormValues}): Promise<{data: AppCallResponse<any>}> => {
+        //TODO use FormResponseData instead of Any
+        const {form} = this.state;
+        if (!form) {
+            return makeError('submitForm state.form is not defined');
+        }
+
+        const call = this.getCall();
+        if (!call) {
+            return makeError('submitForm props.call is not defined');
+        }
+
+        try {
+            const res = await this.props.actions.doAppCall({
+                ...call,
+                type: AppCallTypes.SUBMIT,
+                values: submission.values,
+            });
+            return res;
+        } catch (e) {
+            return {data: {type: 'error', error: e.message}};
+        }
+    }
+
+    refreshOnSelect = async (field: AppField, values: AppFormValues, value: AppFormValue): Promise<{data: AppCallResponse<any>}> => {
+        const {form} = this.state;
+        if (!form) {
+            return makeError('refreshOnSelect props.form is not defined');
+        }
+
+        const call = this.getCall();
+        if (!call) {
+            return makeError('refreshOnSelect props.call is not defined');
+        }
+
+        if (!field.refresh) {
+            return {
+                data: {
+                    type: '',
+                    data: {},
+                },
+            };
+        }
+
+        try {
+            const res = await this.props.actions.doAppCall({
+                ...call,
+                type: AppCallTypes.FORM,
+                values: {
+                    name: field.name,
+                    values,
+                    value,
+                },
+            });
+
+            if (res?.data?.form) {
+                this.setState({form: res.data.form});
+            }
+
+            return res;
+        } catch (e) {
+            return {data: {type: 'error', error: e.message}};
+        }
+    };
+
+    performLookupCall = async (field: AppField, formValues: AppFormValues, userInput: string): Promise<AppSelectOption[]> => {
+        const call = this.getCall();
+        if (!call) {
+            return [];
+        }
+
+        const values: AppLookupCallValues = {
+            name: field.name,
+            user_input: userInput,
+            values: formValues,
+        };
+        const res = await this.props.actions.doAppCall({
+            ...call,
+            type: AppCallTypes.LOOKUP,
+            values,
+        });
+
+        if (res.data.type === AppCallResponseTypes.ERROR) {
+            return [];
+        }
+
+        const data = res.data.data as {items: AppSelectOption[]};
+        if (data.items && data.items.length) {
+            return data.items;
+        }
+
+        return [];
+    }
+
+    getCall = (): AppCall | null => {
+        const {form} = this.state;
```

Comment:
check for `form` like below (line 159)?

Reply:
Theoretically, `getCall` does not need a valid form. An undefined form also would work.